src/engine/fileOpsIns.cpp

```diff
@@ -810,16 +1008,148 @@ void DivEngine::loadFF(SafeReader& reader, std::vector<DivInstrument*>& ret, Str
 }
 
 void DivEngine::loadOPM(SafeReader& reader, std::vector<DivInstrument*>& ret, String& stripPath) {
-  DivInstrument* ins[128];
-  memset(ins,0,128*sizeof(void*));
+  std::vector<DivInstrument*> insList;
+
+  int readCount = 0;
+
+  bool patchNameRead = false,
+       lfoRead = false,
+       characteristicRead = false,
+       m1Read = false,
+       c1Read = false,
+       m2Read = false,
+       c2Read = false;
+
+  DivInstrument* newPatch = nullptr;
+  
+  auto completePatchRead = [&]() {
+    return patchNameRead && lfoRead && characteristicRead && m1Read && c1Read && m2Read && c2Read;
+  };
+  auto resetPatchRead = [&]() {
+    patchNameRead = lfoRead = characteristicRead = m1Read = c1Read = m2Read = c2Read = false;
+    newPatch = nullptr;
+  };
+  auto readIntStrWithinRange = [](String&& input, int limitLow, int limitHigh) {
+    int x = atoi(input.c_str());
```

Comment:
std::stoi

You may use exception handling (try/catch). It's more reliable than atoi.

Reply:
Cool, happy for parser to be more nazi with OPM params.